### PR TITLE
Upgraded AngleSharp to v0.11.0

### PR DESCRIPTION
**Pull request by j055**
Changed .NET Framework support to v4.6 and above. Fixed code to work with changed AngleSharp API.
Fixed one test.

### CODE REVIEW

**Comment by martinnormark** [comment]
Thanks for this!

Is AngleSharp the main reason for dropping support for .Net Framework 4.5?

**Comment by j055** [comment]
Yes it is.

**Comment by martinnormark** [comment]
Alright, other PRs was merged this morning. Do you want to update your fork from `master` and verify that your changes still works?

**Comment by j055** [comment]
Done, I think!

File: PreMailer.Net/PreMailer.Net/PreMailer.cs [hunk 1]
```diff
@@ -29,18 +28,9 @@ public class PreMailer : IDisposable
...
-		/// <summary>
-		/// Constructor for the PreMailer class
-		/// </summary>
-		/// <param name="htmlDoc">The HTML document.</param>
-		/// <param name="baseUri">Url that all relative urls will be off of</param>
-		public PreMailer(IHtmlDocument htmlDoc, Uri baseUri = null)
 		{
 			_baseUri = baseUri;
-			_document = htmlDoc;
+            _document = new HtmlParser().ParseDocument(html);
```
Comment (martinnormark) [inline PreMailer.Net/PreMailer.Net/PreMailer.cs]: This is reverting changes in #169

File: PreMailer.Net/PreMailer.Net/PreMailer.cs [hunk 2]
```diff
@@ -411,24 +394,22 @@ private void StripElementAttributes(params string[] attributeNames)
...
-
-		/// <summary>
-		/// Dispose underlying document
-		/// </summary>
-		public void Dispose()
-		{
-			_document.Dispose();
-		}
-	}
+        /// <summary>
```
Comment (martinnormark) [inline PreMailer.Net/PreMailer.Net/PreMailer.cs]: Reverts changes from #169

File: PreMailer.Net/PreMailer.Net/PreMailer.cs [hunk 3]
```diff
@@ -184,14 +174,7 @@ private IEnumerable<string> GetCssBlocks(IEnumerable<ICssSource> cssSources)
...
 			{
-                try
-                {
-                    styleBlocks.Add(styleSource.GetCss());
-                }
-                catch (System.Net.WebException e)
-                {
-                    _warnings.Add(e.Message);
-                }
+				styleBlocks.Add(styleSource.GetCss());
```
Comment (martinnormark) [inline PreMailer.Net/PreMailer.Net/PreMailer.cs]: Will exceptions be handled here gracefully? I wonder if we still need try-catch block?

File: PreMailer.Net/PreMailer.Net/PreMailer.cs [hunk 4]
```diff
@@ -129,7 +119,7 @@ public InlineResult MoveCssInline(bool removeStyleElements = false, string ignor
 				}
 			}
 
-			var html = _document.ToHtml(new AutoSelectedMarkupFormatter(_document.Doctype));
+			var html = _document.ToHtml();
```
Comment (martinnormark) [inline PreMailer.Net/PreMailer.Net/PreMailer.cs]: Are there breaking changes in AngleSharp update here?

In #169, you can pass an `IHtmlDocument` instance to PreMailer, and before we used `DocType` as formatting when producing HTML.

Can we maintain that, or isn't formatting options supported now?

**Comment by j055** [comment]
Looks like I messed something up with my merge. I'll have another go.

File: PreMailer.Net/PreMailer.Net/PreMailer.cs (hunk 4 above)
Comment (j055) [reply PreMailer.Net/PreMailer.Net/PreMailer.cs]: `AutoSelectedMarkupFormatter` doesn't exist in 0.11.0.

**Comment by petrosmm** [comment]
Hello, is there any plans to create with a new release with this commit soon?

**Comment by reegeek** [comment]
Any update on this PR ?

**Pull request merged**